application/core/utils/fileExtensions.php

```diff
@@ -0,0 +1,785 @@
+	'image/x-cmx'=>'cmx',
+	'image/x-freehand'=>'fh',
+	'image/x-icon'=>'ico',
+	'image/x-mrsid-image'=>'sid',
+	'image/x-pcx'=>'pcx',
+	'image/x-pict'=>'pic',
+	'image/x-portable-anymap'=>'pnm',
+	'image/x-portable-bitmap'=>'pbm',
+	'image/x-portable-graymap'=>'pgm',
+	'image/x-portable-pixmap'=>'ppm',
+	'image/x-rgb'=>'rgb',
+	'image/x-tga'=>'tga',
+	'image/x-xbitmap'=>'xbm',
+	'image/x-xpixmap'=>'xpm',
+	'image/x-xwindowdump'=>'xwd',
+	'message/rfc822'=>'eml',
+	'model/iges'=>'igs',
+	'model/mesh'=>'msh',
+	'model/vnd.collada+xml'=>'dae',
+	'model/vnd.dwf'=>'dwf',
+	'model/vnd.gdl'=>'gdl',
+	'model/vnd.gtw'=>'gtw',
+	'model/vnd.mts'=>'mts',
+	'model/vnd.vtu'=>'vtu',
+	'model/vrml'=>'wrl',
+	'model/x3d+binary'=>'x3db',
+	'model/x3d+vrml'=>'x3dv',
+	'model/x3d+xml'=>'x3d',
+	'text/cache-manifest'=>'appcache',
+	'text/calendar'=>'ics',
+	'text/css'=>'css',
+	'text/csv'=>'csv',
+	'text/html'=>'html',
+	'text/n3'=>'n3',
+	'text/plain'=>'txt',
+	'text/prs.lines.tag'=>'dsc',
+	'text/richtext'=>'rtx',
+	'text/sgml'=>'sgml',
+	'text/tab-separated-values'=>'tsv',
+	'text/troff'=>'t',
+	'text/turtle'=>'ttl',
+	'text/uri-list'=>'uri',
+	'text/vcard'=>'vcard',
+	'text/vnd.curl'=>'curl',
+	'text/vnd.curl.dcurl'=>'dcurl',
+	'text/vnd.curl.scurl'=>'scurl',
+	'text/vnd.curl.mcurl'=>'mcurl',
+	'text/vnd.dvb.subtitle'=>'sub',
+	'text/vnd.fly'=>'fly',
+	'text/vnd.fmi.flexstor'=>'flx',
+	'text/vnd.graphviz'=>'gv',
+	'text/vnd.in3d.3dml'=>'3dml',
+	'text/vnd.in3d.spot'=>'spot',
+	'text/vnd.sun.j2me.app-descriptor'=>'jad',
+	'text/vnd.wap.wml'=>'wml',
+	'text/vnd.wap.wmlscript'=>'wmls',
+	'text/x-asm'=>'s',
+	'text/x-c'=>'c',
+	'text/x-fortran'=>'f',
+	'text/x-java-source'=>'java',
+	'text/x-opml'=>'opml',
+	'text/x-pascal'=>'p',
+	'text/x-nfo'=>'nfo',
+	'text/x-setext'=>'etx',
+	'text/x-sfv'=>'sfv',
+	'text/x-uuencode'=>'uu',
+	'text/x-vcalendar'=>'vcs',
+	'text/x-vcard'=>'vcf',
+	'video/3gpp'=>'3gp',
+	'video/3gpp2'=>'3g2',
+	'video/h261'=>'h261',
+	'video/h263'=>'h263',
+	'video/h264'=>'h264',
+	'video/jpeg'=>'jpgv',
+	'video/jpm'=>'jpm',
+	'video/mj2'=>'mj2',
+	'video/mp4'=>'mp4',
+	'video/mpeg'=>'mpg',
+	'video/ogg'=>'ogv',
+	'video/quicktime'=>'mov',
+	'video/vnd.dece.hd'=>'uvh',
+	'video/vnd.dece.mobile'=>'uvm',
+	'video/vnd.dece.pd'=>'uvp',
+	'video/vnd.dece.sd'=>'uvs',
+	'video/vnd.dece.video'=>'uvv',
+	'video/vnd.dvb.file'=>'dvb',
+	'video/vnd.fvt'=>'fvt',
+	'video/vnd.mpegurl'=>'mxu',
+	'video/vnd.ms-playready.media.pyv'=>'pyv',
+	'video/vnd.uvvu.mp4'=>'uvu',
+	'video/vnd.vivo'=>'viv',
+	'video/webm'=>'webm',
+	'video/x-f4v'=>'f4v',
+	'video/x-fli'=>'fli',
+	'video/x-flv'=>'flv',
+	'video/x-m4v'=>'m4v',
+	'video/x-matroska'=>'mkv',
+	'video/x-mng'=>'mng',
+	'video/x-ms-asf'=>'asf',
+	'video/x-ms-vob'=>'vob',
+	'video/x-ms-wm'=>'wm',
+	'video/x-ms-wmv'=>'wmv',
+	'video/x-ms-wmx'=>'wmx',
+	'video/x-ms-wvx'=>'wvx',
+	'video/x-msvideo'=>'avi',
+	'video/x-sgi-movie'=>'movie',
+	'video/x-smv'=>'smv',
+	'x-conference/x-cooltalk'=>'ice',
+	/* LimeSurvey */
```

Comment:
Copy of framework file , adding this 4 lines.

Reply:
If you do `include` you don't have to copy the entire file, and it could still support changes made to the external library.

Reply:
fileExtensions.php is 
````
return array(

);
````

Nothing else,

You want it like that :
````
$mimeTypes = array_merge(
    require(Yii::getPathOfAlias('system.utils.fileExtensions').'.php'), // Yii framework
    require(Yii::getPathOfAlias('application.core.utils.fileExtensionsLimesurvey').'.php'), // LimeSurvey
);
```` in `LSFileHelper.php`

Or 

````
return array_merge(
    require(Yii::getPathOfAlias('system.utils.fileExtensions').'.php'), // Yii framework
    array(
        'image/heif'=>'heif',
        'image/heif-sequence'=>'heifs',
        'image/heic'=>'heic',
        'image/heic-sequence'=>'heics',
    )
);````  in `fileExtensions.php`

?

Reply:
Solution 2 seems better (IMO)

Reply:
Both seem OK to me, you pick. :)

Reply:
`/* iphone image : issue #15624 */`

Adding commit id with each array can be an idea ;)

Reply:
I tested it. Its not working with .heic files.